File: apps/opsAgent/app/contracts/operationContract.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


OperationAction = Literal["START", "STOP", "RESTART"]
OperationStatus = Literal["SUCCESS", "FAILED"]
ServiceKey = Literal["backend", "studentWeb"]


class OperationRequestContract(BaseModel):
    contractName: Literal["OperationRequest.v1"] = "OperationRequest.v1"
    action: OperationAction
    target: str


class OperationalEventContract(BaseModel):
    contractName: Literal["OperationalEvent.v1"] = "OperationalEvent.v1"
    eventId: str
    action: OperationAction
    target: str
    requestedAt: datetime
    finishedAt: datetime
    status: OperationStatus
    affectedServices: list[str] = Field(default_factory=list)
    errorType: str | None = None


class ManagedServiceStatusContract(BaseModel):
    contractName: Literal["ManagedServiceStatus.v1"] = "ManagedServiceStatus.v1"
    serviceKey: ServiceKey
    containerName: str
    state: Literal["RUNNING", "STOPPED", "MISSING", "UNKNOWN"]


class OperationsStatusContract(BaseModel):
    contractName: Literal["OperationsStatus.v1"] = "OperationsStatus.v1"
    services: list[ManagedServiceStatusContract]

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/controlCenter/src/contracts/operationContract.ts
export type OperationAction = 'START' | 'STOP' | 'RESTART'
export type OperationStatus = 'SUCCESS' | 'FAILED'
export type ManagedServiceState = 'RUNNING' | 'STOPPED' | 'MISSING' | 'UNKNOWN'
export type ManagedServiceKey = 'backend' | 'studentWeb'

export interface OperationalEventContract {
  contractName: 'OperationalEvent.v1'
  eventId: string
  action: OperationAction
  target: string
  requestedAt: string
  finishedAt: string
  status: OperationStatus
  affectedServices: string[]
  errorType?: string | null
}

export interface ManagedServiceStatusContract {
  contractName: 'ManagedServiceStatus.v1'
  serviceKey: ManagedServiceKey
  containerName: string
  state: ManagedServiceState
}

export interface OperationsStatusContract {
  contractName: 'OperationsStatus.v1'
  services: ManagedServiceStatusContract[]
}
